flask_appbuilder/security/sqla/apis/user/user.py

```diff
@@ -0,0 +1,200 @@
+from datetime import datetime
+
+from flask import current_app
+from flask import g, request
+from flask_appbuilder import ModelRestApi
+from flask_appbuilder.api import expose, safe
+from flask_appbuilder.const import API_RESULT_RES_KEY
+from flask_appbuilder.models.sqla.interface import SQLAInterface
+from flask_appbuilder.security.decorators import permission_name, protect
+from flask_appbuilder.security.sqla.models import Role, User
+from marshmallow import ValidationError
+from sqlalchemy.exc import IntegrityError
+from werkzeug.security import generate_password_hash
+
+from .schema import UserPostSchema, UserPutSchema
+
+
+class UserApi(ModelRestApi):
+    resource_name = "user"
+    class_permission_name = "User"
+    datamodel = SQLAInterface(User)
+    allow_browser_login = True
+
+    list_columns = [
+        "id",
+        "roles",
+        "first_name",
+        "last_name",
+        "username",
+        "active",
+        "email",
+        "last_login",
+        "login_count",
+        "fail_login_count",
+        "created_on",
+        "changed_on",
+        "created_by.id",
+        "changed_by.id",
+        "created.id",
+        "changed.id",
+    ]
+    show_columns = list_columns
+    add_columns = [
+        "roles",
+        "first_name",
+        "last_name",
+        "username",
+        "active",
+        "email",
+        "password",
+    ]
+    edit_columns = add_columns
+    search_columns = list_columns
+
+    add_model_schema = UserPostSchema()
+    edit_model_schema = UserPutSchema()
+
+    def pre_update(self, item):
+        item.changed_on = datetime.now()
+        item.changed_by_fk = g.user.id
```

Comment:
isn't this getting handled by the Audit mixin?

Reply:
the default `User` model doesn't use Auditmixin and have `changed_on` column defined without `onupdate` property.
so had to add this.

Reply:
no, your right